python/grass/benchmark/plots.py

```diff
@@ -54,18 +54,28 @@ def nprocs_plot(results, filename=None, title=None):
     Each result can come with a different list of nprocs, i.e., benchmarks
     which used different values for nprocs can be combined in one plot.
     """
+    ylabel = ""
     plt = get_pyplot(to_file=bool(filename))
     axes = plt.gca()
 
     x_ticks = set()  # gather x values
     for result in results:
         x = result.nprocs
         x_ticks.update(x)
-        plt.plot(x, result.times, label=result.label)
-        if hasattr(result, "all_times"):
+        if metric == "time":
             mins = [min(i) for i in result.all_times]
             maxes = [max(i) for i in result.all_times]
+            plt.plot(x, result.times, label=result.label)
             plt.fill_between(x, mins, maxes, color="gray", alpha=0.3)
+            ylabel = "Time [s]"
+        elif metric in ["speedup", "efficiency"]:
+            ylabel = metric.title()
+            plt.plot(x, getattr(result, metric), label=result.label)
```

Comment:
Why using `getattr`?

Reply:
Because `metric` is a variable and `result` is not a dictionary, we cannot use `result.metric` or `result[metric]`.

Reply:
Right, makes sense.